### DISCUSSION TITLE
【Master分支编译】be启动报Failed to initialize JNI: Failed to find the library %s.

### DISCUSSION THREAD

**Discussion by action2013** [apache/doris#14110] [RFC]
根据文档[Doris编译](https://doris.apache.org/zh-CN/docs/install/source-install/compilation)使用win11的wsl安装docker编译doris，编译指令为：WITH_MYSQL=1 WITH_LZO=1 sh build.sh --clean --be --fe

编译成功后拷贝到centos7.6，启动be提示如下错误：
![image](https://user-images.githubusercontent.com/7157818/200764621-83497622-b0ba-40f8-ba72-f4a5ca0fcc0c.png)

错误信息：
Failed to initialize JNI: Failed to find the library %s.
的%s没有正确显示是哪个库加载失败。

**Comment by action2013** [comment]
分析源码发现错误来自：
![image](https://user-images.githubusercontent.com/7157818/200766173-36ffb0d7-bf2e-4c59-b6ef-dbdbc23ef153.png)

**Comment by action2013** [comment]
错误原因为JAVA_HOME环境变量找不到
![image](https://user-images.githubusercontent.com/7157818/200768474-2116dcf9-9c6c-4f48-9a4a-66c017c0b937.png)

但实际是存在的：
![image](https://user-images.githubusercontent.com/7157818/200768509-703d2ede-be19-4224-9ab1-ba482dc4ae4f.png)

而且LIBJVM_SO=libjvm.so也存在：
![image](https://user-images.githubusercontent.com/7157818/200768846-1075ceb0-0de4-4e90-94e8-f004a44b111f.png)

**Comment by action2013** [comment]
最后修改 start_be.sh 添加 JAVA_HOME环境变量后能正常启动：
![image](https://user-images.githubusercontent.com/7157818/200773430-eebe0927-9c9c-442d-a02d-4516b6cc9338.png)

目前还不知道为什么无法自动获取这个环境变量

**Comment by zy-kkk** [comment]
你好，请执行一下show variables where variable_name = 'version_comment';看一下版本
还有请问一下，当前问题和[doris1.1.3使用mysql外部表be奔溃](https://github.com/apache/doris/discussions/14118)是一个问题吗

**Reply by action2013** [reply]
1. 我添加了WITH_MYSQL导致了使用了原生mysql外部和odbc外部导致的，我已在评论补充说明了。
2. 跟[Failed to initialize JNI: Failed to find the library %s.](https://github.com/apache/doris/discussions/14110#top)所使用的也是自己编译的那个版本
3. ![image](https://user-images.githubusercontent.com/7157818/200831441-49b80601-bd9a-4ca2-abf0-88fd922eed32.png) 看版本号好像不是1.1.3，这个是不是我切换分支没正确切换到1.1.3导致的？
4. 下载官方的1.1.3的版本号是： 
![image](https://user-images.githubusercontent.com/7157818/200832805-2e5815f2-1c0a-4343-9461-caa7c00fbcf5.png)
5. 我用git branch查了一下当前是master分支：
![image](https://user-images.githubusercontent.com/7157818/200834897-50f03128-a8ee-442e-94a1-80f5b00116b9.png)

**Reply by zy-kkk** [reply]
你的编译分支是master，由于在master支持了新的jdbc外表，所以be所在的机器需要Java环境，请正确配置，然后需要注意，be的lib目录相比之前多了java-udf-jar-with-dependencies.jar这个jar包，所以建议升级时替换整个lib目录
需要注意的的是，master分支不是稳定版本，还请切换正确的分支（1.1-lts）或者1.1的tag来编译，我们近期会发布基于master分支的1.2版本。
还有一点，如果你现在的集群是基于旧的1.1.x升级到你自己编译的master，请不要降级，会导致元数据错误，如果之前有备份元数据，可以基于备份的元数据重新部署1.1.3，如果没有，需要清空元数据再重新部署

**Reply by action2013** [reply]
好的，多谢
